pkg/common/security/usergroup_test.go

```diff
@@ -162,38 +225,78 @@ func TestCacheCleanUp(t *testing.T) {
 		t.Error("Lookup should not have failed: testuser2 user")
 	}
 
+	testCache.lock.Lock()
 	ug := testCache.ugs["testuser1"]
 	if ug.failed {
 		t.Error("User 'testuser1' not resolved as a success")
 	}
 	// expire the successful lookup
 	ug.resolved -= 2 * poscache
+	testCache.lock.Unlock()
 
 	// resolve a non existing user
 	_, err = testCache.GetUserGroup("unknown")
 	if err == nil {
 		t.Error("Lookup should have failed: unknown user")
 	}
+	testCache.lock.Lock()
 	ug = testCache.ugs["unknown"]
 	if !ug.failed {
 		t.Error("User 'unknown' not resolved as a failure")
 	}
 	// expire the failed lookup
 	ug.resolved -= 2 * negcache
+	testCache.lock.Unlock()
 
 	testCache.cleanUpCache()
-	if len(testCache.ugs) != 1 {
-		t.Errorf("Cache not cleaned up : %v", testCache.ugs)
-	}
+	assert.Equal(t, 1, testCache.getUGsize(), "Cache is not empty: %v", testCache.getUGmap())
+}
+
+func TestIntervalCacheCleanUp(t *testing.T) {
+	testCache := GetUserGroupCache("test")
+	testCache.resetCache()
+	// test cache should be empty now
+	assert.Equal(t, 0, testCache.getUGsize(), "Cache is not empty: %v", testCache.getUGmap())
+
+	// resolve an existing user
+	_, err := testCache.GetUserGroup("testuser1")
+	assert.NilError(t, err, "Lookup should not have failed: testuser1 user")
+
+	_, err = testCache.GetUserGroup("testuser2")
+	assert.NilError(t, err, "Lookup should not have failed: testuser1 user")
+
+	testCache.lock.Lock()
+	ug := testCache.ugs["testuser1"]
+	assert.Assert(t, !ug.failed, "User 'testuser1' not resolved as a success")
+	testCache.lock.Unlock()
+
+	// expire the successful lookup
+	testCache.lock.Lock()
+	ug.resolved -= 2 * poscache
+
+	testCache.lock.Unlock()
```

Comment:
Resolve the `UserGroup` object for "testuser1" here:

```
	// expire the successful lookup
	testCache.lock.Lock()
	ug := testCache.ugs["testuser1"]
	ug.resolved -= 2 * poscache
	testCache.lock.Unlock()
```

This way it's cleaner and more obvious what you're modifying.